### DISCUSSION TITLE
Add BS Tooltips

### DISCUSSION THREAD

**Discussion by AidanBSH** [bootscore/bootscore#254] [Q&A] [answered]
Morgen Basti
Can you confirm that this is possible? 
My plan was to add popper.js to the custom.js file in the child theme
I guess I also need to implement the global customization by adding `$('[data-toggle="popover"]').popover()`. Guessing again but add it to the _bscore_variables.scss file.

By the way sass compiler also around 45 seconds on a Hetzner server here in Germany.

Grüße aus den Schwarzwald  Aidan

**Comment by crftwrk** [comment, marked as answer]
Hi AidanBSH,

- https://bootscore.me/theme/cheatsheet/#tooltips
- https://bootscore.me/theme/cheatsheet/#popovers

`popper.js` is included in https://github.com/bootscore/bootscore/blob/main/js/lib/bootstrap.bundle.min.js. Tooltips and Popovers are disabled in Bootstrap by default for performance reason and you must init them by yourself  **inside** the `jQuery(function ($) {` ... `});` wrapper in your child's `custom.js`. It has nothing to do with `_bscore_variables.scss`.

```JS
// Tooltips
var tooltipTriggerList = [].slice.call(document.querySelectorAll('[data-bs-toggle="tooltip"]'))
var tooltipList = tooltipTriggerList.map(function (tooltipTriggerEl) {
    return new bootstrap.Tooltip(tooltipTriggerEl)
})

// Popovers
var popoverTriggerList = [].slice.call(document.querySelectorAll('[data-bs-toggle="popover"]'))
var popoverList = popoverTriggerList.map(function (popoverTriggerEl) {
    return new bootstrap.Popover(popoverTriggerEl)
})
```

> By the way sass compiler also around 45 seconds on a Hetzner server here in Germany.

Please check here https://github.com/bootscore/bootscore/issues/252

Does this help?

**Reply by AidanBSH** [reply]
Thanks - took me 5 minutes to find the correct procedure to mark as answered!!

**Comment by AidanBSH** [comment]
Thanks for prompt reply will try it later and revert if need more help.

Had already read #252. Not sure my IT knowledge is up to changing anything in server settings to improve so will live with it.

Great product by the way it is the third Bootstrap theme that I have tried to integrate with wordpress and appears the very best.
Aidan

**Reply by AidanBSH** [reply]
Works great thanks.
![image](https://user-images.githubusercontent.com/113333312/189677208-b5790e76-226d-45d1-949e-af17485e32c2.png)

**Reply by crftwrk** [reply]
👍 You're welcome. Please mark the solution as answered.

> Not sure my IT knowledge is up to changing anything in server settings to improve so will live with it.

You do not need any IT knowledge, all settings can be made in hoster's backend. I think Hetzner as quiet good support as well, ask there how to optimize your server.
